Update multi-cluster user guide and templates

luolanzone opened the pull request:
Add a quick start guide.
Update the user guide with Multi-cluster Gateway.

Signed-off-by: Lan Luo <luola@vmware.com>
Signed-off-by: Jianjun Shen <shenj@vmware.com>

jianjuns commented:
@luolanzone : I updated the quick-start doc. The biggest change is that it now talks about two clusters, with one to be both leader and member. Could you check? I plan to revise the user guide too.

jianjuns commented on docs/multicluster/quick-start.md:
```diff
@@ -0,0 +1,172 @@
+
+In this quick start guide, we will set up a Antrea Multi-cluster ClusterSet with
+two clusters. One cluster will serve as both the leader and a member cluster,
+while another cluster will be a member only. The diagram belows shows the two
```
@luolanzone : I feel we better to have a simple diagram to show the two clusters (with cluster ID), and ClusterSet (with leader and member cluster information).

luolanzone replied:
Sure, I haven't had the chance to draw the diagram yesterday, I will add it.

jianjuns replied:
A few comments for the diagram:

1. What you think we have a box for the ClusterSet that contains the two clusters? We can add ClusterSet ID there.
2. Add something like "leader" to the leader mc-controller to indicate it is the controller for leader.
3. And some way to indicate the Gateway Node too.
4. Could we add apiserver IP to Cluster A
5. Could we indicate Cluster A is "leader and member", and Cluster B is "member"

luolanzone commented on multicluster/config/samples/clusterset_init/multicluster_clusterset_template.yaml:
```diff
@@ -1,15 +1,15 @@
 apiVersion: multicluster.crd.antrea.io/v1alpha1
 kind: ClusterClaim
 metadata:
-  name: leadercluster-id
```
yes, I will change this one today.

luolanzone replied:
Changed it to `clusterset.k8s.io` due to upper case is not allowed in metadata's name.

jianjuns commented on docs/multicluster/user-guide.md:
````diff
@@ -312,57 +334,123 @@ Then create below `ClusterClaim` and `ClusterSet` in the cluster `test-cluster-n
+  namespace: antrea-multicluster
 ```
 
+## Multi-cluster Gateway Configuration
````
I moved it here to be before the feature consumption sections. But have not review and update the section yet.

jianjuns commented on docs/multicluster/user-guide.md:
````diff
@@ -503,6 +591,17 @@ Events:
   Warning ACNPImportFailed     2m11s  resourceimport-controller  ACNP Tier random does not exist in the importing cluster test-cluster-west
 ```
 
+## Build Antrea Multi-cluster Image
````
This is not really for users. But anyway, I moved it to the end of the doc.

jianjuns commented on docs/multicluster/user-guide.md:
```diff
@@ -2,99 +2,114 @@
+cluster, one in leader mode and another in member mode. There is no deployment
+dependency between member and leader clusters when you have a dedicated leader
+cluster. But to run both leader and member controllers in one cluster, you need
+to follow the deployment sequence below:
```
Is it still true? @luolanzone

luolanzone replied:
I think there is no dependency even the cluster run both leader and member if user is using 'kubectl apply', but if user is using `kubectl create -f`, CRD installation will fail since some of them are already existing like `ClusterClaim`.

jianjuns replied:
If it is true, we should remove the words around ordering.

luolanzone commented on multicluster/config/samples/clusterset_init/multicluster_membercluster_template.yaml:
```diff
@@ -25,5 +25,6 @@ spec:
-    - clusterID: test-cluster-east
-  namespace: antrea-mcs-ns
+    - clusterID: test-cluster-leader
+    - clusterID: test-cluster-member
```
@hjiajing  Could you help to review these changes in templates? I hope it has minor impact to CI, thanks.

jianjuns replied:
@luolanzone : please update the ObjectMeta.Name first.

luolanzone replied:
sure, I  am working the diagram, will submit a new commit once is done.

jianjuns commented on docs/multicluster/quick-start.md:
````diff
@@ -0,0 +1,172 @@
+
+```bash
+$kubectl apply -f https://github.com/antrea-io/antrea/releases/download/$TAG/antrea-multicluster-member.yml
+$kubectl apply -f https://github.com/antrea-io/antrea/releases/download/$TAG/antrea-multicluster-leader-global.yml
````
Do we need to apply antrea-multicluster-leader-global.yml for a dual-role cluster? In the user guide, earlier you wrote it is necessary, as antrea-multicluster-member.yml already import all CRDs.

If we do need to apply all YAMLs and the order does not matter, I would apply leader YAMLs first, which is easier to understand. @luolanzone

luolanzone replied:
yes, we removed those unnecessary CRDs in leader yamls, so we need to apply antrea-multicluster-leader-global.yml for a dual-role cluster, I will move it.

luolanzone replied:
Done, the diagram is also updated.

jianjuns reviewed (commented):
Re-paste my comments for the diagram:

1. What you think we have a box for the ClusterSet that contains the two clusters? We can add ClusterSet ID there.
2. Add something like "leader" to the leader mc-controller to indicate it is the controller for leader.
3. And some way to indicate the Gateway Node too.
4. Could we add apiserver IP to Cluster A
5. Could we indicate Cluster A is "leader and member", and Cluster B is "member"

jianjuns commented on docs/multicluster/quick-start.md:
````diff
@@ -0,0 +1,175 @@
+
+```bash
+$kubectl apply -f https://github.com/antrea-io/antrea/releases/download/$TAG/antrea-multicluster-leader-global.yml
+$kubectl apply -f https://github.com/antrea-io/antrea/releases/download/$TAG/antrea-multicluster-member.yml
````
I noticed you added this line in your last update, but antrea-multicluster-member.yml is already applied in line 60. Should we remove this line?

jianjuns replied:
I also revised the diagram a little. Check.

jianjuns replied:
Ok.

jianjuns commented on docs/multicluster/quick-start.md:
````diff
@@ -0,0 +1,174 @@
+$sed 's/changeme/antrea-multicluster/g' antrea-multicluster-leader-namespaced.yml | kubectl apply -f -
+$kubectl apply -f https://github.com/antrea-io/antrea/releases/download/$TAG/antrea-multicluster-member.yml
+```
+
````
It might be good to add a command to check the deployment results that can show both leader & member mc-controller Deployments and Pods.

Does something like work: `kc get all -A -l="app=xyz"`

What should be the "xyz".

luolanzone replied:
sure, I will add it, it can be `kubectl get all -A -l="component=antrea-mc-controller"`, should I add a sample output?

luolanzone replied:
Done

hjiajing commented:
@luolanzone Hi Lan. I changed `multiclusterv1alpha1.WellKnownClusterClaimClusterSet` from `clusterSet.k8s.io` to `clusterset.k8s.io`. I don't know if this change will affect you in this PR. Just a reminder.

luolanzone commented:
@hjiajing , I will check, thanks for the reminder, please make sure e2e is passed after you change the ID.

luolanzone commented:
Hi @jianjuns  I also changed the `WellKnownClusterClaimClusterSet` from  "clusterSet.k8s.io"` "clusterset.k8s.io" in this PR and all other related strings, maybe be do a rebase later if @hjiajing 's antctl bootstrap is merged first. and I changed your sign-off info to co-author in the commit, otherwise, there will be no extra co-author information.

jianjuns commented:
/skip-all
